ChaosMod/Components/Globals.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "Components/Component.h"
+
+#include "Memory/Hooks/ScriptThreadRunHook.h"
+
+#include <map>
+#include <vector>
+
+enum GlobalPatternIdiom : uint8_t
+{
+	GLOBAL_U16,
+	GLOBAL_U24
+};
+
+struct GlobalRegistration
+{
+  public:
+	std::string m_Name;
+	std::string m_ScriptName;
+	std::string m_Pattern;
+	int m_Offset;
+	GlobalPatternIdiom m_PatternIdiom;
+
+	GlobalRegistration(std::string name, std::string scriptName, std::string pattern, int offset,
+	                   GlobalPatternIdiom patternIdiom)
+	    : m_Name(name), m_ScriptName(scriptName), m_Pattern(pattern), m_Offset(offset), m_PatternIdiom(patternIdiom)
+	{
+	}
+};
+
+class Globals : public Component
+{
+  private:
+	static inline std::map<std::string, int> m_GlobalsIndexMap = {};
```

Comment:
```suggestion
	static inline std::map<std::string, int> ms_GlobalsIndexMap = {};
```
(Also needs to be changed everywhere else)